packages/core/src/domain/tracekit/tracekit.ts

```diff
@@ -55,36 +57,44 @@ export function startUnhandledErrorCollection(callback: UnhandledErrorCallback)
  */
 function instrumentOnError(callback: UnhandledErrorCallback) {
   return instrumentMethodAndCallOriginal(window, 'onerror', {
-    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: Error) {
-      let stack: StackTrace
-
-      if (errorObj) {
-        stack = computeStackTrace(errorObj)
+    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: unknown) {
+      // Priority is given to errorObj if it is an instance of Error. This allows us to compute the most comprehensive stack trace.
+      if (errorObj instanceof Error) {
+        const stack = computeStackTrace(errorObj)
         callback(stack, errorObj)
       } else {
-        const location = {
-          url,
-          column: columnNo,
-          line: lineNo,
-        }
+        let name: string | undefined
+        let msg: string | undefined
 
-        let name
-        let msg = message
+        // Try to extract information about the error from the provided message
         if ({}.toString.call(message) === '[object String]') {
-          const groups = ERROR_TYPES_RE.exec(msg as string)
+          msg = message as string
+          const groups = ERROR_TYPES_RE.exec(message as string)
           if (groups) {
             name = groups[1]
             msg = groups[2]
           }
         }
 
-        stack = {
-          name,
-          message: typeof msg === 'string' ? msg : undefined,
-          stack: [location],
+        // If an errorObj is provided, give priority to its content to populate the message of the stack trace
+        if (errorObj) {
+          msg = jsonStringify(errorObj)
         }
 
-        callback(stack, message)
+        const stack: StackTrace = {
+          name,
+          // We only include the `Uncaught` prefix when errorObj is present, to preserve backward compatibility
+          message: errorObj && msg ? `${NonErrorPrefix.UNCAUGHT} ${msg}` : msg,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** It could be nice to limit `trackit` responsibilities to `Cross-browser collection of unhandled errors` only and not introducing our logic of building error messages here.